### PR TITLE
fix cache corruption when shrink multiple times

### PR DESCRIPTION
**Pull request by superyyrrzz**
If multiple outputs' content are same in cache, shrink can remove duplicate files and make these items point to the same file.

**Before**: First shrink per version, then shrink per PostProcessor in all versions. The second shrink can find duplicates cross versions, but will make the existing cache reference invalid, as version manifest has already been written to disk.

**this fix**: shrink only once before write version manifest on disk.

### CODE REVIEW

**Comment by fenxu** [comment]
@vwxyzh @vicancy Please help to take a look if convenient.

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/BuildInfo.cs [hunk 1]
```diff
@@ -108,6 +108,16 @@ public static BuildInfo Load(string baseDir, bool onlyValid)
             return buildInfo;
         }
 
+        public void SaveVersionsManifet(string baseDir)
+        {
+            var expanded = Path.GetFullPath(Environment.ExpandEnvironmentVariables(baseDir));
+            var targetDirectory = Path.Combine(expanded, DirectoryName);
```
Comment (partychen) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/BuildInfo.cs]: Seems there is no used for targetDirectory

**Comment by herohua** [comment]
do we have test case for non-versioned case?

**Comment by superyyrrzz** [comment]
@herohua The added case in this PR also have 2 identical outputs within a version.

**Pull request merged**